x-pack/plugins/cloud_security_posture/server/fleet_integration/fleet_integration.ts

```diff
@@ -79,17 +81,28 @@ export const onPackagePolicyPostCreateCallback = async (
 export const onPackagePolicyDeleteCallback = async (
   logger: Logger,
   deletedPackagePolicy: ArrayElement<DeletePackagePoliciesResponse>,
-  soClient: ISavedObjectsRepository
+  soClient: ISavedObjectsRepository,
+  taskManager: TaskManagerStartContract,
+  taskId: string
 ): Promise<void> => {
   try {
     const { saved_objects: cspRules }: SavedObjectsFindResponse<CspRuleSchema> =
       await soClient.find({
         type: cspRuleAssetSavedObjectType,
         filter: `${cspRuleAssetSavedObjectType}.attributes.package_policy_id: ${deletedPackagePolicy.id} AND ${cspRuleAssetSavedObjectType}.attributes.policy_id: ${deletedPackagePolicy.policy_id}`,
+        perPage: 10000,
       });
     await Promise.all(
       cspRules.map((rule) => soClient.delete(cspRuleAssetSavedObjectType, rule.id))
     );
+    const { saved_objects: postDeleteRules }: SavedObjectsFindResponse<CspRuleSchema> =
```

Comment:
you can check if we are installed by checking if the plugin is installed. (checkout plugin.ts)